### ISSUE TITLE
Update sample code to also test REST transport

### ISSUE THREAD

**Issue by parthea** [googleapis/python-bigquery-datatransfer#440]
PR #439 added support for using REST transport. We should also update the handwritten sample code [here](https://github.com/googleapis/python-bigquery-datatransfer/tree/main/samples/snippets) to support both gRPC and REST transport.

**Comment by tswast** [comment]
Some words of caution: 

BQ-DTS defaults to a very limited quota of like 4 API calls per minute or something tiny like that. Adding REST samples may double the quota used.

Even when we updated the samples tests to use a per-Python version project in https://github.com/googleapis/synthtool/pull/778, it was somehow still using up the Kokoro build pool project's quota until I hacked the quota project config in the "transfer_client" fixture in https://github.com/googleapis/python-bigquery-datatransfer/pull/83/files

At least at the time of https://github.com/googleapis/python-bigquery-datatransfer/pull/83, there was very limited quota and not much capacity available when we asked for more. It has been three years, so we might be able to revisit that request for more quota.
